python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -339,17 +340,46 @@ def evaluate(self, *, cache: MutableMapping[int, DataFrame]) -> DataFrame:
                 colnames[0],
             )
         elif self.typ == "parquet":
-            tbl_w_meta = plc.io.parquet.read_parquet(
-                plc.io.SourceInfo(self.paths),
-                columns=with_columns,
-                nrows=n_rows,
-                skip_rows=self.skip_rows,
-            )
-            df = DataFrame.from_table(
-                tbl_w_meta.tbl,
-                # TODO: consider nested column names?
-                tbl_w_meta.column_names(include_children=False),
-            )
+            if parquet_options.get("chunked", True):
+                reader = plc.io.parquet.ChunkedParquetReader(
+                    plc.io.SourceInfo(self.paths),
+                    columns=with_columns,
+                    num_rows=n_rows,
+                    skip_rows=self.skip_rows,
+                    chunk_read_limit=parquet_options.get("chunk_read_limit", 0),
+                    pass_read_limit=parquet_options.get("pass_read_limit", 1024000000),
+                )
+                chk = reader.read_chunk()
+                tbl = chk.tbl
+                names = chk.column_names()
+                concatenated_columns = tbl.columns()
+                while reader.has_next():
+                    tbl = reader.read_chunk().tbl
+
+                    for i in range(tbl.num_columns()):
+                        concatenated_columns[i] = plc.concatenate.concatenate(
+                            [concatenated_columns[i], tbl._columns[i]]
+                        )
+                        # Drop residual columns to save memory
+                        tbl._columns[i] = None
```

Comment:
This would do fewer copies if we did something like:
```
tables = []
chunk = reader.read_chunk()
tables.append(chunk.tbl)
names = chunk.column_names()
while reader.has_next():
   tables.append(reader.read_chunk().tbl)
# deliberately overwriting so we free the chunk tables
tables = plc.concatenate.concatenate(tables)
df = DataFrame.from_table(tables, names=names)
```

Reply:
It would be really nice if `reader` had a `column_names` method, because then we could do:
```
def tables(reader):
    while reader.has_next():
        yield reader.read_chunk().tbl
names = reader.column_names()
df = DataFrame.from_table(plc.concatenate(list(tables(reader))), names=names)
```
and then we wouldn't need to worry about manually releasing objects.

Reply:
@brandon-b-miller can you summarize the results of all the concatenation experiments here so that we have a record of what decision was made and why?

Reply:
Yes - basically here we tried varying the way we concat things together in the end from less concatting of larger chunks to more concatting of smaller ones and now lean towards the second approach. Larger concats appear to provide a marginal performance improvement while also introducing significant instability at larger dataset sizes which didn't seem to be worth it. Something like a few % higher throughput for large datasets at the expense of OOM-ing a bunch more elsewhere.